src/axom/core/examples/core_utilities.cpp

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*! \file core_utilities.cpp
+ *  \brief This example code is a demonstration of the Axom Core utilites.
+ */
+
+ /* This example code contains snippets used in the Primal Sphinx documentation.
+  * They begin and end with comments such as
+  *
+  * timer_start
+  * timer_end
+  *
+  * each prepended with an underscore.
+  */
+
+
+#ifdef WIN32
+#include "windows.h"
+void sleep(int numSeconds)
+{
+  int numMilliSecs = numSeconds * 1000;
+  Sleep(numMilliSecs);
+}
+#else
+#include <unistd.h> // for sleep()
+#endif
+
+// C/C++ includes
+#include <iostream>
+#include <vector>
+
+// Axom includes
+// _header_start
+#include "axom/core/utilities/FileUtilities.hpp"
+#include "axom/core/utilities/StringUtilities.hpp"
+#include "axom/core/utilities/Timer.hpp"
+#include "axom/core/utilities/About.hpp"
+// _header_end
+
+// _fs_string_start
+void demoFileSystemAndString(const char* argv0)
+{
+  using namespace axom::utilities;
+
+  // Get the current directory
+  std::string cwd = filesystem::getCWD();
+
+  // Split it on file separator.
+#if WIN32
+  const char pathsep = '\\';
+#else
+  const char pathsep = '/';
+#endif
+  std::vector<std::string> cmp;
+  string::split(cmp, cwd, pathsep);
+
+  // Count how many start with "ax" or end with "exe"
+  // (we could also use std::count_if)
+  int matchcount = 0;
+  std::string prefix{ "ax" };
+  std::string suffix{ "exe" };
+  for (int i = 0; i < cmp.size(); ++i)
+  {
+    if (string::startsWith(cmp[i], prefix) || 
+        string::endsWith(cmp[i], suffix))
+    {
+      matchcount += 1;
+    }
+  }
+  std::cout << "Found " << matchcount << " path components starting with " << 
+    prefix << " or ending with " << suffix << "." << std::endl;
+
+  // Append "hello.txt"
+  std::string hellofile = 
+    filesystem::joinPath(cwd, "hello.txt", std::string{pathsep});
```

Comment:
outside of this PR... but shouldn't this function handle the pathsep internally...

Reply:
It really should